packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -290,61 +296,64 @@ async function uploadFile(
     'uploadConcurrency',
     uploadOptions.uploadConcurrency
   )
-  const maxChunkSize = utils.assertDefined(
-    'uploadChunkSize',
-    uploadOptions.uploadChunkSize
-  )
+  const maxChunkSize = 32 * 1024 * 1024 // 32 MB Chunks
 
-  const parallelUploads = [...new Array(concurrency).keys()]
+  const parallelUploads = presignedUrls
   core.debug('Awaiting all uploads')
-  let offset = 0
 
+  let etags: string[] = []
   try {
-    await Promise.all(
-      parallelUploads.map(async () => {
-        while (offset < fileSize) {
-          const chunkSize = Math.min(fileSize - offset, maxChunkSize)
-          const start = offset
-          const end = offset + chunkSize - 1
-          offset += maxChunkSize
-
-          await uploadChunk(
-            httpClient,
-            resourceUrl,
-            () =>
-              fs
-                .createReadStream(archivePath, {
-                  fd,
-                  start,
-                  end,
-                  autoClose: false
-                })
-                .on('error', error => {
-                  throw new Error(
-                    `Cache upload failed because file read failed with ${error.message}`
-                  )
-                }),
-            start,
-            end
-          )
-        }
+    etags = await Promise.all(
+      parallelUploads.map(async (url, index) => {
+        const offset = index * maxChunkSize;
+        const chunkSize = Math.min(fileSize - offset, maxChunkSize)
+        const start = offset
+        const end = offset + chunkSize - 1
+
+        core.debug(`Uploading #${index} chunk url:${url}: start:${start} end:${end} size:${fileSize}`)
```

Comment:
Do you want to keep this for the initial version?

Reply:
It only displays when debug mode is enabled. It doesn't print logs under normal circumstances.